core/src/main/scala/akka/kafka/internal/CommittingProducerSinkStage.scala

```diff
@@ -167,6 +167,7 @@ private final class CommittingProducerSinkStageLogic[K, V, IN <: Envelope[K, V,
 
   override protected def onTimer(timerKey: Any): Unit = timerKey match {
     case CommittingProducerSinkStage.CommitNow => commit(Interval)
+    case _ =>
```

Comment:
would be good to throw to avoid potential future foot-shooting/head-scratching

Reply:
Right, I added more logging/errors for unexpected matches.